### PR TITLE
Add an option to reject the request if binding and body value has conflict.

### PR DESCRIPTION
**Pull request by shuoyang2016**

### CODE REVIEW

File: src/request_weaver.cc [hunk 1]
```diff
@@ -31,10 +35,131 @@ namespace transcoding {
...
+                         const pbconv::DataPiece& value_in_body,
+                         const pbconv::DataPiece& value_in_binding) {
+  bool value_is_same = true;
+  switch (value_in_body.type()) {
+    case pbconv::DataPiece::TYPE_INT32: {
+      pb::util::StatusOr<int32_t> status = value_in_binding.ToInt32();
+      if (!status.ok()) {
+        return bindingFailureStatus(field_name, "int32", value_in_binding);
+      }
+      if (status.value() != value_in_body.ToInt32().value())
```
Comment (qiwzhang) [inline src/request_weaver.cc:60]: I remember c++ style guild requires 

if (expr) {
   one_line_code;
}

File: src/request_weaver.cc [hunk 2]
```diff
@@ -229,7 +366,7 @@ void RequestWeaver::CollisionCheck(internal::string_view name) {
 
 RequestWeaver::WeaveInfo* RequestWeaver::WeaveInfo::FindWeaveMsg(
     const internal::string_view field_name) {
-  for (auto& msg : messages) {
+  for (auto& msg: messages) {
```
Comment (qiwzhang) [inline src/request_weaver.cc:369]: against style guide?

File: test/request_weaver_test.cc [hunk 3]
```diff
@@ -46,17 +48,20 @@ class RequestWeaverTest : public ::testing::Test {
     std::vector<std::string> field_names =
         absl::StrSplit(field_path_str, ".", absl::SkipEmpty());
     std::vector<const Field*> field_path;
-    for (const auto& n : field_names) {
+    for (const auto& n: field_names) {
```
Comment (qiwzhang) [inline test/request_weaver_test.cc:51]: new style?

**Review by qiwzhang — APPROVED** [review]
Just some minor style questions

File: src/request_weaver.cc (hunk 2 above)
Comment (shuoyang2016) [reply src/request_weaver.cc]: Reverted. It's auto changed by my clion.

File: test/request_weaver_test.cc (hunk 3 above)
Comment (shuoyang2016) [reply test/request_weaver_test.cc]: Reverted. It's auto changed by my clion.

File: src/request_weaver.cc (hunk 1 above)
Comment (shuoyang2016) [reply src/request_weaver.cc]: SG. Added parenthesis for this line and other lines as well.

**Review by qiwzhang — APPROVED** [review]

**Pull request merged**